tools/print_deps_for_target.sh

```diff
@@ -0,0 +1,46 @@
+#!/bin/bash
+
+set -o errexit
+set -o pipefail
+
+
+if [[ -z "$(which jq)" ]]; then
+  echo "jq is required"
+  exit 1
+fi
+
+readonly target_dep_output=$(mktemp)
+readonly repos=$(mktemp)
+readonly recipies=$(mktemp)
+readonly all_deps=$(mktemp)
+
+bazel query 'deps('"$1"')' 1> "$target_dep_output" 2> /dev/null
+
+# Capture a list of explicit repositories
+grep "^@" "$target_dep_output" \
+  | awk -F "//" '{ print $1 }' \
+  | sort | uniq \
+  | cut -d@ -f2 1> "$repos"
+
+# Capture a list of implicit repositories (build recipies)
+grep "^@envoy_deps" "$target_dep_output" \
+  | grep -v  thirdparty \
+  | cut -d: -f2 \
+  | sort | uniq 1> "$recipies"
+
+# Resolve the recipe repository name to the dependency name
+cat "$recipies" | python -c "
+import imp
+import sys
+
+MAP=imp.load_source('deps', 'bazel/target_recipes.bzl')
+
+print('\n'.join([MAP.TARGET_RECIPES[recipe.strip()] for recipe in sys.stdin]))
+" | sort | uniq >> "$repos"
+
+./tools/print_dependencies.py 1> "$all_deps"
+
+cat "$repos" \
```

Comment:
How do you feel about rewriting this script in Python? Upsides include not needing to install `jq`, less `awk` and bash magic, no need to generate Python code... Downside is you probably have a script that works reasonable already.

Reply:
I thought about that but didn't want to shell out to run `bazel query`

Reply:
`subprocess.check_output` gives a single line "invoke this shell command, ensure it returns 0 and give me the output as a string".